Cython/Compiler/Nodes.py

```diff
@@ -2789,8 +2789,9 @@ def error_value(self):
         if self.return_type.is_pyobject:
             return "0"
         else:
-            #return None
-            return self.entry.type.exception_value
+            # protect against the corner-case where the user has specified a c++ translation function
+            if not isinstance(self.entry.type.exception_value, Node):
```

Comment:
I'm not hugely keen on this, but couldn't find a better way of doing it

Reply:
I wonder if this should use a `NameNode` instead. Special cases are not special enough to break the rules.

Reply:
Is this actually needed? According to your comments, if it's a function name, then it's a `NameNode` already, right?
(Note that your change leaves the method without return value in that case).